### DISCUSSION TITLE
Allow PREFIX support in lotus Makefile

### DISCUSSION THREAD

**Discussion by Erotemic** [filecoin-project/lotus#8499] [Ideas] [answered]
I like to build and install packages into my $HOME/.local folder so I can differentiate system packages from ones that I built myself.

I'm looking at the lotus Makefile and I see that it hard codes the install PREFIX to /usr/local

```
.PHONY: build

install: install-daemon install-miner install-worker

install-daemon:
	install -C ./lotus /usr/local/bin/lotus

install-miner:
	install -C ./lotus-miner /usr/local/bin/lotus-miner

install-worker:
	install -C ./lotus-worker /usr/local/bin/lotus-worker

install-app:
	install -C ./$(APP) /usr/local/bin/$(APP)
```

Is there a reason for this? Would adding a patch for a PREFIX be a welcome addition?

**Comment by LexLuthr** [comment, marked as answer]
@Erotemic You can simply copy out the built binaries to any directory you like. There is no need to install it at all. 
Currently, install is tied to the systemd config and adding a prefix would over complicate things for new users. Experienced users can simply copy out the binaries after building them without installation.

**Reply by Erotemic** [reply]
My thought is that if you are reading a Makefile you probably know what a PREFIX is, and inexperienced users would either (1) use the snap distribution or (2) run the instructions with all of the defaults. I don't think there is any over-complication in conforming to [Makefile standards](https://www.gnu.org/prep/standards/html_node/Directory-Variables.html). 

The workaround you've proposed is fine though, it would just be nice if the standard PREFIX construct worked out of the box.

**Reply by LexLuthr** [reply]
@Erotemic Can you file the request with details? 
https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml

**Comment by LexLuthr** [comment]
I am archiving this discussion as a feature request can be filed here.
